Kernel/Memory/Region.cpp

```diff
@@ -472,6 +472,27 @@ PageFaultResponse Region::handle_inode_fault(size_t page_index_in_region)
     auto page_index_in_vmobject = translate_to_vmobject_page(page_index_in_region);
     auto& vmobject_physical_page_slot = inode_vmobject.physical_pages()[page_index_in_vmobject];
 
+    {
+        // Note: We don't acquire the inode_vmobject Spinlock because:
+        // 1. It will be meaningless because we don't mutate the VMObject
+        // 2. Calling inode_vmobject.inode().size() might require acquiring a mutex
+        //  of that inode. The Ext2FSInode::metadata() does that, for example.
+        //  Holding that lock while we try to lock this Spinlock will lead to
+        //  deadlocks as other processes might need to lock it, but we will not be able
+        //  to preempt under this Spinlock being held.
+        //
+        // Instead, we hold the Inode mutex of the InodeVMObject, to ensure
+        // we get the correct size, in both calls to inode_vmobject.inode().size(),
+        // so the Inode cannot change its size in that time.
+        MutexLocker locker(inode_vmobject.inode().inode_lock());
```

Comment:
instead of doing that, can't we just fetch the size once instead of twice and use that?